File: LeetCode/python3/1002.py
class Solution:
    def commonChars(self, A: List[str]) -> List[str]:
        dict = {}
        for c in range(ord('a'), ord('z') + 1):
            dict[chr(c)] = sys.maxsize
        for a in A:
            temp = {}
            for c in a:
                if c not in temp:
                    temp[c] = 1
                else:
                    temp[c] += 1
            for c in dict:
                if c not in temp:
                    dict[c] = 0
                else:
                    dict[c] = min(dict[c], temp[c])
        ret = []
        for c in dict:
            for i in range(dict[c]):
                ret.append(c)
        return ret

File: LeetCode/cpp/1002.cpp
class Solution {
public:
    vector<string> commonChars(vector<string> &A) {
        int n = A.size();
        if (n == 0)
            return {};
        vector<vector<int>> chars(n, vector<int>(26, 0));
        for (int i = 0; i < n; ++i) {
            string a = A[i];
            for (char c:a)
                ++chars[i][c - 'a'];
        }
        vector<int> res = chars[0];
        for (int i = 1; i < n; ++i) {
            for (int j = 0; j < chars[i].size(); ++j)
                res[j] = min(res[j], chars[i][j]);
        }
        vector<string> ret;
        for (int i = 0; i < res.size(); ++i) {
            for (int j = 0; j < res[i]; ++j)
                ret.emplace_back(string(1, i + 'a'));
        }
        return ret;
    }
};